From 7342d3405765d0c39c91e801d2bd8856901205ce Mon Sep 17 00:00:00 2001
From: Wu Liang feng <wulf@rock-chips.com>
Date: Wed, 17 Aug 2016 12:02:04 +0800
Subject: [PATCH] arm64: dts: rockchip: add usb3 controller reset for rk3399

We can assert the reset to keep the whole USB3 Controller
in resetting to hold pipe power state in P2 before
initializing Type-C PHY.

Change-Id: Ibb5716bac645ae01ee27fd019a3dfcbd3c0ffd84
Signed-off-by: Wu Liang feng <wulf@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3399.dtsi | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399.dtsi b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
index 709970d7191a..5a351966e008 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
@@ -439,6 +439,8 @@
 			      "aclk_usb3otg0", "aclk_usb3_rksoc_axi_perf",
 			      "aclk_usb3", "aclk_usb3_grf";
 		power-domains = <&power RK3399_PD_USB3>;
+		resets = <&cru SRST_A_USB3_OTG0>;
+		reset-names = "usb3-otg";
 		#address-cells = <2>;
 		#size-cells = <2>;
 		ranges;
@@ -468,6 +470,8 @@
 			      "aclk_usb3otg1", "aclk_usb3_rksoc_axi_perf",
 			      "aclk_usb3", "aclk_usb3_grf";
 		power-domains = <&power RK3399_PD_USB3>;
+		resets = <&cru SRST_A_USB3_OTG1>;
+		reset-names = "usb3-otg";
 		#address-cells = <2>;
 		#size-cells = <2>;
 		ranges;
-- 
2.35.3

